recipes/m4/all/conanfile.py

```diff
@@ -116,8 +130,3 @@ def package_info(self):
         self.runenv_info.define_path("M4", m4_bin)
         self.buildenv_info.define_path("M4", m4_bin)
 
-        # TODO: to remove in conan v2
-        bin_path = os.path.join(self.package_folder, "bin")
-        self.output.info(f"Appending PATH environment variable: {bin_path}")
-        self.env_info.PATH.append(bin_path)
-        self.env_info.M4 = m4_bin
```

Comment:
no, it breaks conan v1. conan-center recipes don't have to handle bugs in conan v2 beta.

Reply:
Hi @SpaceIm
Thanks for your feedback. I don't think this is a bug in Conan 2 but rather a compatibility issue with v1. I'm trying a new change that hopefully works with both. Thanks

Reply:
it's a v1 compatibility issue of conan v2-beta client, so a bug in conan client, and it has been decided that recipes won't be fixed but conan client v2-beta when such issue occurs.